### PR TITLE
feat(protocol-designer): add custom liquid color picker

### LINKED ISSUE #10573: Liquid Setup in PD: add custom liquid color picker
# Overview

We need to add custom liquid color picker to the new color field in Protocol Designer. 

# Implementation details

PD will pre-select a liquid color based on the order in the [color picker design](https://www.figma.com/file/2xjSLHXquNsKvC1mV9NMjr/Milestone-2?node-id=955%3A145056), but users will also be able to click the color field and update it too. 

Color picker functionality should use and follow: https://casesandberg.github.io/react-color/

Here is a loom recording for more information on how custom liquid colors work in PD: https://www.loom.com/share/b13830b0267b4e69827cfb346c8e238d

# Design

https://www.figma.com/file/2xjSLHXquNsKvC1mV9NMjr/Milestone-2?node-id=955%3A144974

# Acceptance criteria
- [x] Add feature flag in PD for liquid color enhancements
- [x] Add new colors for PD liquids
- [x] Color field updates to new color when that color is selected in the picker
- [x] Hex code field can be edited by user in the color picker 
- [x] Hex code updates when pre-defined liquid color is selected
- [x] When the user clicks the hex code input field they get the blinking cursor as it's a text input field
- [x] PD pre-selects a liquid color when it is added based on the order of the colors in the color picker design
- [x] Selected color only applies to liquids list when user selects save

### LINKED ISSUE #10601: Liquid Setup in PD: add liquid color field
# Overview

We need to add a liquid color field to the liquids tab in Protocol Designer.

# Implementation details

This will be a dynamic field based on the color picker work outlined in [this ticket](https://app.zenhub.com/workspaces/app--ui-5f6248dba845b000139aa767/issues/opentrons/opentrons/10573).

# Design

https://www.figma.com/file/2xjSLHXquNsKvC1mV9NMjr/Milestone-2?node-id=955%3A144974

# Acceptance criteria

- [x] "Color" field appears on liquid setup page
- [x] Color field hover state renders the same as other input fields
- [x] Color picker window opens when color field clicked on

### PR DESCRIPTION
**Pull request by sakibh**
# Overview

This PR adds a custom liquid color picker and field to the liquids page in Protocol Designer. closes #10573 , closes #10601 

Co-authored by @smb2268 

image

Note: still working on updating/adding tests

# Changelog

- Added `react-color`
- Added `displayColor` field to liquids page component and redux store
- Updated PD liquids colors

# Review requests

Note: In order to access this new feature, enable the feature flag called `Enable liquid color enhancements`.

- Check that the ACs for both tickets mentioned above are fulfilled.
- Check that you are able to choose any of the predefined colors AND/OR add a custom color via hex code.
- Check that the colors are persisting once you click save.
- Check that when adding liquid to labware, the colors are accurate/persistent.
- Check that when uploading an old protocol, PD prompts you to update the version and the liquid colors should default to new the new color scheme.

# Risk assessment

low, behind FF

### CODE REVIEW

**Comment by mmencarelli** [comment]
hey @sakibh is there any way for me to test this in PD? If not, could you send over a video? Thanks!

**Comment by sakibh** [comment]
> hey @sakibh is there any way for me to test this in PD? If not, could you send over a video? Thanks!

Hey @mmencarelli, you can access this branch for PD here: http://sandbox.designer.opentrons.com/pd_add-color-picker/

**Review by mmencarelli — CHANGES_REQUESTED** [review]
Wow, this is amazing! 

K, the only change i can see here is that you have the hover state as the enabled state. So move the box shadow over to only appear on hover!

File: protocol-designer/src/labware-ingred/selectors.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ import max from 'lodash/max'
 import reduce from 'lodash/reduce'
 import { Options } from '@opentrons/components'
 import { LabwareLiquidState } from '@opentrons/step-generation'
+// import { swatchColors } from '../components/swatchColors'
```
Comment (smb2268) [inline protocol-designer/src/labware-ingred/selectors.ts]: looks like this can be removed

**Review by smb2268 — APPROVED** [review]
tested this out and it works & looks great!

**Comment by mmencarelli** [comment]
Could we have the box-shadow appear when the menu is active/open? 

Reference to design [on Figma](https://www.figma.com/file/2xjSLHXquNsKvC1mV9NMjr/Milestone-2?node-id=955%3A145056).

Other than that, looks great!

**Review by mmencarelli — APPROVED** [review]

**Pull request merged**